app/src/main/java/com/example/drawingApp/MainActivity.kt

```diff
@@ -1,23 +1,118 @@
 package com.example.drawingApp
 
 import android.os.Bundle
+import android.os.StrictMode
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
 import androidx.activity.viewModels
+import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.isVisible
+import androidx.core.widget.addTextChangedListener
 
 class MainActivity : AppCompatActivity() {
-    private val model: MainViewModel by viewModels()
+    private val model: DrawingViewModel by viewModels()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-
+        enableStrictMode()
         val colorCircleView = findViewById<SelectedColorView>(R.id.colorCircle)
         val drawingFieldView = findViewById<DrawingFieldView>(R.id.drawField)
 
+        drawingFieldView.onLineDrawing = {
+            model.activeBitmap = it
+        }
+
+        findViewById<TextView>(R.id.newLayer).setOnClickListener {
+            showLayerAlertDialog()
+        }
+
+        model.activeBitmap?.let {
+            drawingFieldView.setBitmap(it)
+            drawingFieldView.invalidate()
+        }
+
         colorCircleView.onColorChange = { stateColor ->
             model.primaryColor = stateColor
             drawingFieldView.updateColor(stateColor)
         }
 
+        if (model.isLayerCreationDialogOpen) showLayerAlertDialog()
+
+    }
+
+    fun showLayerAlertDialog() {
+        val inflater = LayoutInflater.from(this@MainActivity)
+        val view = inflater.inflate(R.layout.layer_dialog, null)
+        val layerField = view.findViewById<EditText>(R.id.newLayerField)
+        val errorText = view.findViewById<TextView>(R.id.dialogCharCountError)
+        var layerCounter = 1
+
+        //find the first available layer name not taken
+        while (model.layerNames.contains(getString(R.string.new_layer_hint, layerCounter))) {
+            layerCounter += 1
+        }
+        layerField.setText(getString(R.string.new_layer_hint, layerCounter))
+
+        //needs to be after the set text or else the textchanged listener will overwrite the previous text needed for rotations
+        layerField.addTextChangedListener {
+            model.currentLayerCreationDialogText = layerField.text.toString()
+            errorText.isVisible = it.toString().trim().length > MAX_CHAR_LEN
+            if (errorText.isVisible) {
+                layerField.setText(layerField.text.toString().dropLast(1))
+                //visibility needs to be after or else the dropLast will make it never visible
+                errorText.visibility = View.VISIBLE
```

Comment:
Why are you changing its visibility when you already set its visibility on line 64?

Reply:
because of the droplast it will never show the error text without the visibility being set there

Reply:
now instead of a droplast it is a substing but same thought applies.

Reply:
This doesn't make sense. You are setting its visibility, using it in an if statement, and then setting it again using a different method. For one, errorText will ALWAYS be visible here. You are confirming so with your if statement. What part am I not seeing?